src/libraries/System.Text.Json/gen/JsonSourceGenerator.Parser.cs

```diff
@@ -81,109 +84,286 @@ public Parser(Compilation compilation)
+                    Debug.Assert(contextTypeSymbol != null);
+
+                    contextGenSpecList ??= new List<ContextGenerationSpec>();
+                    contextGenSpecList.Add(new ContextGenerationSpec
+                    {
+                        SerializerOptions = options ?? new JsonSerializerOptionsAttribute(),
+                        ContextType = contextTypeSymbol.AsType(_metadataLoadContext),
+                        RootSerializableTypes = rootTypes,
+                    });
+
+                    // Clear the cache of generated metadata between the processing of context classes.
+                    _typeGenerationSpecCache.Clear();
+                }
+
+                if (contextGenSpecList == null)
+                {
+                    return null;
+                }
+
+                return new SourceGenerationSpec
+                {
+                    ContextGenerationSpecList = contextGenSpecList,
+                    BooleanType = _booleanType,
+                    ByteArrayType = _byteArrayType,
+                    CharType = _charType,
+                    DateTimeType = _dateTimeType,
+                    DateTimeOffsetType = _dateTimeOffsetType,
+                    GuidType = _guidType,
+                    StringType = _stringType,
+                    NumberTypes = _numberTypes,
+                };
+            }
+
+            // Returns true if a given type
+            // - derives from JsonSerializerContext
+            // - is partial
+            // - has no members
+            private bool IsConfigurableJsonSerializerContext(
+                ClassDeclarationSyntax classDeclarationSyntax,
+                INamedTypeSymbol jsonSerializerContextSymbol,
+                SemanticModel compilationSemanticModel)
+            {
+                SeparatedSyntaxList<BaseTypeSyntax>? baseTypeSyntaxList = classDeclarationSyntax.BaseList?.Types;
+                if (baseTypeSyntaxList == null)
+                {
+                    return false;
+                }
+
+                INamedTypeSymbol? match = null;
+
+                foreach (BaseTypeSyntax baseTypeSyntax in baseTypeSyntaxList)
+                {
+                    INamedTypeSymbol? candidate = compilationSemanticModel.GetSymbolInfo(baseTypeSyntax.Type).Symbol as INamedTypeSymbol;
+                    if (candidate != null && jsonSerializerContextSymbol.Equals(candidate, SymbolEqualityComparer.Default))
+                    {
+                        match = candidate;
+                        break;
+                    }
+                }
+
+                if (match == null)
+                {
+                    return false;
+                }
+
+                bool isPartial = classDeclarationSyntax.Modifiers.Where(token => token.IsKind(Microsoft.CodeAnalysis.CSharp.SyntaxKind.PartialKeyword)).FirstOrDefault() != default;
```

Comment:
Do we want to give some diagnostics to the user when they forgot to add `partial` to the class? Currently if I write:

```C#
    [JsonSerializable(typeof(Person))]
    [JsonSerializable(typeof(bool))]
    internal class JsonContext : JsonSerializerContext
    {
```

Nothing gets generated, and I don't have an indication why.

Reply:
Makes sense, done.